### PR TITLE
Cleanup test files copied over the host application

### PR DESCRIPTION
**Pull request by kennyadsl**
## Summary

There were multiple inconsistencies and outdated code. Please read commit by commit for more details. 

## Checklist

Check out our [PR guidelines](https://github.com/solidusio/.github/blob/master/CONTRIBUTING.md#pull-request-guidelines) for more details.

The following are mandatory for all PRs:

- [ ] I have written a thorough PR description.
- [ ] I have kept my commits small and atomic.
- [ ] [I have used clear, explanatory commit messages](https://github.com/solidusio/.github/blob/main/CONTRIBUTING.md#writing-good-commit-messages).

The following are not always needed:

- 📖 I have updated the README to account for my changes.
- 📑 I have documented new code [with YARD](https://www.rubydoc.info/gems/yard/file/docs/Tags.md).
- 🛣️ I have opened a PR to update the [guides](https://github.com/solidusio/edgeguides).
- ✅ I have added automated tests to cover my changes.
- 📸 I have attached screenshots to demo visual changes.

### CODE REVIEW

File: templates/spec/solidus_starter_frontend_spec_helper.rb
Comment (elia) [inline templates/spec/solidus_starter_frontend_spec_helper.rb:88]: This seems to be doing a slightly different thing than the original, probably not an issue, unless it will show up only for some execution sequences.

For 100% compatibility I suggest moving it to a `config.before { Rails.cache.clear }`.

**Review by elia — APPROVED** [review]
All great ✨ 

I left a couple of optional comment, the one about clearing the cache deserving particular attention.

File: templates/spec/solidus_starter_frontend_spec_helper.rb
Comment (waiting-for-dev) [reply templates/spec/solidus_starter_frontend_spec_helper.rb:88]: Or we can move it above the `example.run` line, as we're within an `around(:each)` block.

**Review by waiting-for-dev — APPROVED** [review]
:clap:

File: templates/spec/solidus_starter_frontend_spec_helper.rb
Comment (elia) [reply templates/spec/solidus_starter_frontend_spec_helper.rb:88]: @waiting-for-dev that would probably work as well, although this `around` block is targeting examples with metadata containing `caching: true`.

Comment (kennyadsl) [reply templates/spec/solidus_starter_frontend_spec_helper.rb:88]: > that would probably work as well, although this around block is targeting examples with metadata containing caching: true.

Right, but if the cache is disabled for other specs, does it matter? I think that line where it was before is a bit like shooting a mosquito with a bazooka and we can do better.

**Review by elia — APPROVED** [review]

**Comment by kennyadsl** [comment]
@elia @waiting-for-dev everything addressed, waiting for the CI before merging, thanks!

**Pull request merged**
